server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -251,6 +273,11 @@ void becomeLeader(String method) {
         peerFinder.deactivate(getLocalNode());
         closePrevotingAndElectionScheduler();
         preVoteCollector.update(getPreVoteResponse(), getLocalNode());
+
+        if (leaderCheckScheduler != null) {
```

Comment:
is this necessary? I think we can only transition from candidate to leader. maybe instead assert that it is null?

Reply:
Ok, I pushed 3de59057d9bfcc8731c99d18c8e408bf9ee81390.